test/action/slice.cpp

```diff
@@ -23,22 +23,34 @@ int main()
 {
     using namespace ranges;
 
-    std::vector<int> v = view::ints(0,100);
```

Comment:
I'd prefer that we leave the existing tests untouched - wrap them in `{}`s to scope them out of the way - and only add new ones.

Our jumbled together test style is inexcusable for `action` tests where every line modifies something and we have to go through the test line-by-line to audit anything. (Shhh, don't tell @ericniebler I'm trash-talking his tests.)

Reply:
To be honest i was not happy either with the way i left them either.
Im happy by leaving the old test but im also in favour of fully rewrite them from scratch. If you point me to some good unit test somewhere in this library to pick the style i will get to it.

PS: Dont worry man. I got you. @ericniebler will never know

Reply:
It doesn't need to be anything special; just wrap the individual test cases in block scopes or named functions so they're somewhat encapsulated.